docs/tasks/administer-cluster/ip-masq-agent.md

```diff
@@ -0,0 +1,97 @@
+---
+title: IP Masquerade Agent User Guide
+---
+
+{% capture overview %}
+This page show how to configure and enable the ip-masq-agent. 
+{% endcapture %}
+
+{% capture prerequisites %}
+Kubernetes 1.7
+{% endcapture %}
+
+{% capture discussion %}
+## IP Masquerade Agent User Guide
+
+The ip-masq-agent configures iptables rules to hide a pod's IP address behind the cluster nodes's IP address. This is typically done when sending traffic to destinations outside the cluster's pod [CIDR](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) range.
+
+## **Key Terms**
+
+*   **NAT (Network Address Translation)**
+    *   Is a method of remapping one IP address to another by modifying either the source and/or destination address information in the IP header.  Typically performed by a device doing IP routing
+*   **Masquerading**
+    *   A form of NAT that is typically used to perform a many to one address translation, where multiple source IP addresses are masked behind a single address which is typically the device doing the IP routing - In kubernetes this is the Node's IP address. 
+*   **CIDR (Classless Inter-Domain Routing)**
+    *    Based on the variable-length subnet masking, allows specifying arbitrary-length prefixes. CIDR introduced a new method of representation for IP addresses, now commonly known as **CIDR notation**, in which an address or routing prefix is written with a suffix indicating the number of bits of the prefix, such as 192.168.2.0/24
+*   **Link Local**
+    *   A link-local address is a network address that is valid only for communications within the network segment or the broadcast domain that the host is connected to Link-local addresses for IPv4 are defined in the address block 169.254.0.0/16 in CIDR notation
+
+The ip-masq-agent configures iptables rules to handle masquerading node/pod IP addresses when sending traffic to destinations outside the cluster node's IP and the Cluster IP range.  This essentially hides pod IP addresses behind the cluster node's IP address.  In some environments, traffic to "external" addresses must come from a known machine address. For example, in Google Cloud, any traffic to the internet must come from a VM's IP.  When containers are used, as in GKE, the Pod IP will be rejected for egress. To avoid this, we must hide the Pod IP behind the VM's own IP address - generally known as "masquerade" By default, the agent is configured to treat the three private IP ranges specified by [RFC 1918](https://tools.ietf.org/html/rfc1918) as non-masquerade [CIDR](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).  These ranges are 10.0.0.0/8, 172.16.0.0/12, and 192.168.0.0/16. The agent will also treat link-local (169.254.0.0/16) as a non-masquerade CIDR by default.  The agent is configured to reload its configuration from the location */etc/config/ip-masq-agent* every 60 seconds which is also configurable.
```

Comment:
Missing a period: "...generally known as "masquerade`.`" By default, the agent..."

Add a comma: "...every 60 seconds`,` which is also configurable."